api/app/Controllers/Http/ContactsController.ts

```diff
@@ -0,0 +1,168 @@
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Contact from 'App/Models/Contact'
+import ContactValidator from 'App/Validators/ContactValidator'
+import Logger from '@ioc:Adonis/Core/Logger'
+import { Attachment } from '@ioc:Adonis/Addons/AttachmentLite'
+
+export default class ContactsController {
+
+    public async list ({request, response}: HttpContextContract) {
+        try {
+            const {page, perPage} = request.qs()
+
+            const contacts = await Contact.query().select(['id', 'first_name', 'surname', 'phone_number1', 'company', 'job_title', 'profile_picture']).paginate(page, perPage)
+    
+            return response.ok({data: contacts})
+            
+        } catch (error) {
+            Logger.error('Error at ContactsController.list:\n%o', error)
+            return response.status(error?.status ?? 500).json({
+                message: 'An error occurred while fetching contacts.',
+                error: process.env.NODE_ENV !== 'production' ? error : null
+            })
+        }
+    }
+
+    public async store ({request, response}: HttpContextContract) {
+        try {
+            // const payload = request.body()   // before validation
+            const payload = await request.validate(ContactValidator)
+            const {
+                firstName,
+                surname,
+                jobTitle,
+                email1,
+                email2,
+                phoneNumber1,
+                phoneNumber2,
+                country,
+                streetAddressLine1,
+                streetAddressLine2,
+                city,
+                postCode,
+                state,
+                birthday,
+                website,
+                notes,
+                profilePicture
+            } = payload
```

Comment:
Simplify to:

```ts
            const {
                profilePicture,
                ...contactData
            } = payload
```

Reply:
> Simplify to:
> 
> ```ts
>             const {
>                 profilePicture,
>                 ...contactData
>             } = payload
> ```

I have a question about this sir. In the article, you stated that it is best to destructure each key from the payload for security purposes:
"Since the keys of the JSON data we sent within the request body matches the properties in the Contact model, we could do this:await Contact.create(payload) to quickly persist the new contact to the database. But it is not recommended, and dangerous. This is because malicious actors (aka hackers) could attempt to directly assign values of columns such as id, created_by, and updated_by from the outside. However, these values should be generated by the server. They might not succeed as the API server will override those sensitive value but there is no need to deliberately create that vulnerability.
That is the reason we need to destructure the payload constant and extract the properties we really want to insert into the database. Then at Line 42, we assign those properties to an object which is the only argument to the Contact.create() method which then creates a new row of Contact in our contacts table.  [link to article](https://tech.ndianabasi.com/create-column-definitions-and-insert-new-contacts-or-full-stack-google-contacts-clone-with-adonisjs-nodejs-and-quasar-framework-vuejs)"

That was why we destrutured the payload. If I spread the payload directly in the object in the Contact.create() method,  won't it defeat the purpose of destructuring the payload?

Reply:
The Adonisjs validator removes unknown properties from the payload. So, it is safe to spread the validated payload.

Reply:
> The Adonisjs validator removes unknown properties from the payload. So, it is safe to spread the validated payload.

Okay sir. All changes have been effected